### DISCUSSION TITLE
Jukebox mode not adhering to "Last played older than"

### DISCUSSION THREAD

**Discussion by DamonLane** [jcorporation/myMPD#792] [Q&A] [answered]
First, I agree with [this discussion](https://github.com/jcorporation/myMPD/discussions/65) that Jukebox mode is incredible! I seek out apps that can play "random albums" instead of random or shuffled songs. Other apps use "jukebox" to mean something else so I didn't realize what was here: random song/album/artist _and_ not played recently!? Amazing! 

But I may have one of the several settings wrong because myMPD is repeating albums long before the "older than" input and before it runs out of albums. I have about 550 albums and have the jukebox settings set to:
Mode: Album
Playlist: Database
Keep queue length: 1 (default)
Enforce uniqueness: Album
Last played older than: 500 hours

And it often plays albums it has played in the last few days. Does the queue length have to be longer than the "last played older than" setting?

And what is Autoplay, does that have anything to do with this? Jukebox mode could be called Autoplay and I didn't see definitions of these in the documention, it was reading issues that illuminated the possibilities for me.

**Comment by jcorporation** [comment, marked as answer]
"Last played older than" does not work for the jukebox album mode, it only works for songs. There is no lastPlayed time saved for albums.

Albums are kept unique across queue, jukebox queue and last played. You can increase the `Last played list count` in the settings to increase the timespan for uniq albums.

This behavior is not documented. Commit 17deec9c68c9482b04aab3b903db3b16916daa9e disables the last played older than option if jukebox mode album is selected.

**Comment by DamonLane** [comment]
Thanks for the complete explanation and suggestion of a different way to get a similar effect. Would it be possible to implement this feature using the lastPlayed time for a song in the album?

**Comment by jcorporation** [comment]
Respecting last played for albums based on songs could be a nice feature, but I must look forward how this could be implemented with low overhead.

**Comment by jcorporation** [comment]
This is now implemented in the v9.5.0 feature branch. The jukebox mode does not check if the whole album was last player older than, only a random song from the album.

commit 6c7d5adfa07839defed838523d970036011f6dac

**Comment by DamonLane** [comment]
Fantastic! I look forward to using this, thanks for implementing it so 
quickly!
